integration_tests/tests/test_exposure_schema_validity.py

```diff
@@ -0,0 +1,161 @@
+                "depends_on": {
+                    "nodes": ['models.exposures_test']
+                }
+            }
+        }
+    }
+    test_result = dbt_project.test(
+        test_id,
+        DBT_TEST_NAME,
+        DBT_TEST_ARGS,
+        columns=[dict(
+            name="bla"
+        )],
+        as_model=True
+    )
+    assert test_result["status"] == "pass"
+
+def test_expose_schema_validity_with_correct_columns_and_invalid_type(test_id: str, dbt_project: DbtProject):
+    DBT_TEST_ARGS = {
+        "node": "models.exposures_test",
+        "columns": [
+            {
+                "name": "order_id",
+                "dtype": "numeric"
+            }
+        ],
+        "exposures": {
+            "ZOMG": {
+                "meta": {
+                    "columns": [{
+                        "name": "order_id",
+                        "data_type": "string"
+                    }]
+                },
+                "url": "http://bla.com",
+                "name": "ZOMG",
+                "depends_on": {
+                    "nodes": ['models.exposures_test']
+                }
+            }
+        }
+    }
+    test_result = dbt_project.test(
+        test_id,
+        DBT_TEST_NAME,
+        DBT_TEST_ARGS,
+        columns=[dict(
+            name="bla"
+        )],
+        as_model=True
+    )
+
+    assert 'different data type for the column order_id string vs numeric' in test_result['test_results_query']
+    assert test_result["status"] == "fail"
+
+
+def test_expose_schema_validity_with_correct_columns_and_missing_type(test_id: str, dbt_project: DbtProject):
+    DBT_TEST_ARGS = {
+        "node": "models.exposures_test",
+        "columns": [
+            {
+                "name": "order_id",
+                "dtype": "numeric"
+            }
+        ],
+        "exposures": {
+            "ZOMG": {
+                "meta": {
+                    "columns": [{
+                        "name": "order_id",
+                    }]
+                },
+                "url": "http://bla.com",
+                "name": "ZOMG",
+                "depends_on": {
+                    "nodes": ['models.exposures_test']
+                }
+            }
+        }
+    }
+    test_result = dbt_project.test(
+        test_id,
+        DBT_TEST_NAME,
+        DBT_TEST_ARGS,
+        columns=[dict(
+            name="bla"
+        )],
+        as_model=True
+    )
+
+    assert test_result["status"] == "pass"
+
+def test_expose_schema_validity_with_missing_columns(test_id: str, dbt_project: DbtProject):
+    DBT_TEST_ARGS = {
+        "node": "models.exposures_test",
+        "columns": [
+            {
+                "name": "order",
+                "dtype": "numeric"
+            }
+        ],
+        "exposures": {
+            "ZOMG": {
+                "meta": {
+                    "columns": [{
+                        "name": "order_id",
+                        "data_type": "string"
+                    }]
```

Comment:
I'd discuss also with @ellakz 
but I think each column specified in the exposure should say which model it is from - otherwise I find it confusing if the exposure depends on multiple models.